packages/wallet-sdk/src/core/communicator/Communicator.ts

```diff
@@ -99,7 +100,8 @@ export class Communicator {
       return this.popup;
     }
 
-    this.popup = openPopup(this.url);
+    this.id = `wallet_${crypto.randomUUID()}`;
```

Comment:
I added the change here because I wanted to check the popup name against the id. This is not possible due to browser cross domain security. If this is acceptable as is I plan to move this out of the communicator and into the openPopup function. 

![image](https://github.com/user-attachments/assets/42bd7e65-2f11-486e-9e84-0c94019f7705)

Reply:
i think it should be fine, as we are not really utilizing window name currently at all?